app/views/core/SubscribeModal.coffee

```diff
@@ -114,6 +118,15 @@ module.exports = class SubscribeModal extends ModalView
     #   @startStripeSubscribe()
     @startStripeSubscribe() # Always use Stripe
 
+  onClickYearlyPurchaseButton: (e) ->
+    return unless @basicProductYearly
```

Comment:
How do we end up returning here, and do we need to reset/clean up anything? Give some kind of user friendly error `noty` about annual product somehow missing?

Reply:
I was mimicking the legacy code path with basic monthly subscription. This shouldn't happen since we have validation in the jade to only display the button once the product is loaded.

I am hesitant to remove unless I've misunderstood why it was placed in the legacy path. Thus opting into it being more defensive.